examples/tasks/async_compute.rs

```diff
@@ -0,0 +1,121 @@
+use bevy::{prelude::*, tasks::{AsyncComputeTaskPool, Task}};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use futures_lite::future;
+
+/// This example shows how to use the ecs system and the AsyncComputeTaskPool
+/// to spawn, poll, and complete tasks across systems and system ticks.
+
+// Number of cubes to spawn across the x, y, and z axis
+const NUM_CUBES: i32 = 6;
+
+// Used to tag our new entity spawned with tasks
+struct Marker;
+
+/// This system generates tasks simulating computationally intensive
+/// work that potentially spawns multiple frames/ticks. A separate
+/// system, handle_tasks, will poll the spawned tasks on subsequent
+/// frames/ticks, and use the results to spawn cubes
+fn spawn_tasks(
+    mut commands: Commands,
+    thread_pool: Res<AsyncComputeTaskPool>,
+) {
+    for x in 0 ..NUM_CUBES {
+        for y in 0 ..NUM_CUBES {
+            for z in 0 ..NUM_CUBES {
+
+                // Spawn new task on the AsyncComputeTaskPool
+                let task = thread_pool.spawn(async move {
+
+                    let mut rng = rand::thread_rng();
+                    let start_time = Instant::now();
+                    let duration = Duration::from_secs_f32(rng.gen_range(0.05..0.2));
+                    while Instant::now() - start_time < duration {
+                        // Simulating doing hard compute work generating translation coords!
+                    }
+
+                    // Such hard work, all done!
+                    println!("Done generating translation coords x: {} y: {} z: {}", x, y, z);
+
+                    Transform::from_translation(Vec3::new(x as f32, y as f32, z as f32))
+                });
+
+                // Spawn new entity, tag it with Marker as a component,
+                // and add our new task as a component
+                commands.spawn()
+                    .insert(Marker)
+                    .insert(task);
+            }
+        }
+    }
+}
+
+/// This system queries for entities that have both our Marker component
+/// as well as a Task<Transform> component. It polls the tasks to see if they're
+/// complete. If the task is complete it takes the result, adds a new PbrBundle of components to the
+/// entity using the result from the task's work, and removes the task component from the entity.
+fn handle_tasks(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut our_entity_tasks: Query<(Entity, &mut Task<Transform>), With<Marker>>
+) {
+    for (entity, mut task) in our_entity_tasks.iter_mut() {

```

Comment:
```suggestion
    our_entity_tasks.for_each_mut(|(entity, mut task)| {
```

Reply:
`for_each_mut` has the same functionality as the `for` loop however at slightly better performance. 
It should normally be the default.

Reply:
> `for_each_mut` has the same functionality as the `for` loop however at slightly better performance.
> It should normally be the default.

I prefer this format as well, but if I'm not mistaken, other Bevy examples using Query use the for..in format. I was trying to maintain consistency.

Reply:
Yea, that is something that I'll probably submit a PR to change in the future, however I would strongly prefer the `for_each_mut` so we can set a good example in our examples :)

Reply:
Changed in e31c947